src/Config.php

```diff
@@ -124,6 +126,7 @@ public function setDefault($key, $value)
      * TODO: remove Data object validation in 2.0
      *
      * @return Data
+     * @codeCoverageIgnore
```

Comment:
You shouldn't `@codeCoverageIgnore` here. There is no reason why this shouldn't be covered; if it isn't, that fact should be reflected in the stats and coverage logs.

Reply:
Would you rather just have this uncovered? or do you have a recommendation for how to test this? My research shows that private/protected methods should only be covered by what can be tested publicly. Since these methods are only for use to ensure the `$defaults` property is a `Data` object and not an array, they aren't called by any public methods.

Reply:
I did try subclassing the method to initially create the `$defaults` property as an array, and test using that class, but since that appeared to only test the subclass and didn't change the code coverage for the actual `Config` class.

Unless I'm missing the right way to do this (totally possible 😕 )

Reply:
It's possible to test private / protected methods by using php reflection to set the access level to public, and then call the method indirectly through the reflection object. I do this in some tests (not necessarily in this project, though). That said, I am a fan of testing the just the public methods of a class. If the protected / private methods become really complex and need their own direct testing, that's generally a sign that they should be factored out into another class.

Also, if a protected or private method exists to serve any useful function, then they should be reachable by some public methods, if called with the right input values. If they are never called, that might be an indication that they can be completely removed. Some functions just happen to not be called in tests, e.g. `public function getFoo() { return $this->foo; }` does not necessarily require a test.

These methods look simple enough, and I'm okay with leaving them untested.

Reply:
Ok, thanks. I'll remove the `@codeCoverageIgnore` directives. I think the only part that wasn't getting tested was the parts that validate the `$defaults` property is not an array. I'll try creating a test using reflection that tests this by modifying the `$defaults` property directly and then calls the method.

Reply:
I don't think that you need to use reflection just to call a protected method with bad values to ensure that the sanity checks work. This is testing the specific implementation over the behavior. Either figure out how to cover that code block through the public methods, or leave it untested.

Reply:
I understand your concern with testing implementation instead of behavior, in this case I think it is important to test since these methods are what prevent breaking backward compatibility.

Also in creating the test I found a flaw that I would not have otherwise readily noticed. 😀 

Thanks for merging and for the experience. I have learned a lot through this PR.